fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItem.java

```diff
@@ -90,4 +90,26 @@ default Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers
 	default boolean isSuitableFor(ItemStack stack, BlockState state) {
 		return ((Item) this).isSuitableFor(state);
 	}
+
+	/**
+	 * Returns a leftover item stack after {@code stack} is consumed in a recipe.
+	 * (This is also known as "recipe remainder".)
+	 * For example, using a lava bucket in a furnace as fuel will leave an empty bucket.
+	 *
+	 * <p>This is a stack-aware version of {@link Item#getRecipeRemainder()}.
+	 *
+	 * <p>Reimplementing this method must follow the "best and only practice rule",
+	 * to avoid compatibility issues with other mods using recipe remainders, that
+	 * the returned stack must have a count of 1.
+	 * The only exception is the empty stack that has a count of zero.
+	 *
+	 * <p>For other use cases, is highly recommended to set the remainder via {@link Item.Settings#recipeRemainder(Item)}
+	 * or override {@link Item#getRecipeRemainder()} instead of this.
+	 *
+	 * @param stack the consumed {@link ItemStack}
+	 * @return the leftover item stack
+	 */
```

Comment:
This is the definitive version.

```suggestion
	/**
	 * Returns a leftover item stack after {@code stack} is consumed in a recipe.
	 * (This is also known as "recipe remainder".)
	 * For example, using a lava bucket in a furnace as fuel will leave an empty bucket.
	 *
	 * <p>This is a stack-aware version of {@link Item#getRecipeRemainder()}.
	 *
	 * <p>Overriding this method must follow the rule,
	 * to avoid compatibility issues with other mods using recipe remainders,
	 * that the returned stack must be a new instance with a count of 1.
	 * The only exception is the empty stack that has a count of zero.
	 *
	 * <p>Note that simple item remainders can also be set via {@link Item.Settings#recipeRemainder(Item)}.
	 *
	 * <p>If you want to get a remainder for a stack,
	 * is recommended to use the stack version of this method: {@link FabricItemStack#getRecipeRemainder()}.
	 *
	 * @param stack the consumed {@link ItemStack}
	 * @return the leftover item stack
	 */
```